vb-be/src/modules/socket/socket.gateway.ts

```diff
@@ -7,21 +6,34 @@ import {
   ConnectedSocket,
 } from '@nestjs/websockets';
 import { Server, Socket } from 'socket.io';
-import { WsGuard } from './socket.middleware';
-import { SocketService } from './socket.service';
+
+import { socketError, ErrorText } from 'src/util/error';
+import { HttpStatus } from 'src/util/http';
+
+import { RoomService } from '@modules/room/room.service';
+import { ISocketCreateRoomData } from '@modules/socket/socket.constants';
 
 @WebSocketGateway({ cors: true })
 export class EventsGateway {
   @WebSocketServer()
   server: Server;
 
-  constructor(private readonly socketService: SocketService) {}
+  constructor(private readonly roomService: RoomService) {}
 
   // @UseGuards(WsGuard)
   @SubscribeMessage('create-room')
-  async joinRoom(@MessageBody() data: any, @ConnectedSocket() client: Socket) {}
+  async createRoom(
+    @MessageBody() data: ISocketCreateRoomData,
+    @ConnectedSocket() client: Socket,
+  ) {
+    try {
+      await this.roomService.addRoomToRedis(data.room);
+      client.emit('new-room-added');
+    } catch (err) {
+      return socketError(client, HttpStatus.InternalError, ErrorText.Generic);
```

Comment:
We're going to be facing the same sort of error as we will with HTTP, so I think it'd be easiest for us to implement the same sort of `status` and `message` style of errors for sockets